interface/src/app/tribe_client/directives.js

```diff
@@ -1,80 +1,90 @@
+          UserFactory.getPromise().$promise.then(function() {
+            $scope.userObj = UserFactory.getUser();
+          });
 
-                var modalInstance = $uibModal.open({
-                  templateUrl: 'tribe_client/tribe-login-modal.tpl.html',
-                  controller: [
-                    '$scope', '$uibModalInstance', 'TribeSettings',
-                    function($scope, $uibModalInstance, TribeSettings) {
+          $scope.openLoginModal = function() {
+            var modalInstance = $uibModal.open({
+              templateUrl: 'tribe_client/tribe-login-modal.tpl.html',
+              controller: [
+                '$scope', '$uibModalInstance', 'TribeSettings',
+                function($scope, $uibModalInstance, TribeSettings) {
+                  TribeSettings.get({}, function(data) {
+                    $scope.tribe_url = data['tribe_url'];
+                    $scope.access_code_url = data['access_code_url'];
+                    $scope.tribe_scope = data['scope'];
+                    $scope.client_id = data['client_id'];
+                  });
 
-                      TribeSettings.get({}, function( data ) {
-                          $scope.tribe_url = data['tribe_url'];
-                          $scope.access_code_url = data['access_code_url'];
-                          $scope.tribe_scope = data['scope'];
-                          $scope.client_id = data['client_id'];
-                      });
+                  $scope.refresh = function() {
+                    $rootScope.$broadcast('user.update');
+                    $uibModalInstance.close();
+                    $window.location.reload();
+                  };
 
-                      $scope.refresh = function () {
-                          $rootScope.$broadcast( 'user.update' );
-                          $uibModalInstance.close();
-                          $window.location.reload();
-                      };
-
-                      $scope.cancel = function () {
-                          $uibModalInstance.dismiss('cancel');
-                      };
-                  }]
-                });
-
-                modalInstance.result.then(function (success) {
-                  if (success === true) {
-                      // continue
-                  }
-                  else {
-                  }
-                } , function () {
-                  // This gets called if modal gets dismissed
-                  // when user clicks outside modal, etc.
-                });
-
-            };
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+                }
+              ]
+            });
 
-        }],
-        link: CommonFuncts.updateUser,
-        replace: true,
-        restrict: "E",
-        templateUrl: 'tribe_client/tribe-login-button.tpl.html'
+            modalInstance.result.then(function(success) {
+              if (success === true) {
```

Comment:
Although I am a strong advocate of **===** operator, I am hesitant here: is **success** always a boolean? If it is an object, then this condition will be always false, because **===** doesn't involve any implicit type conversion.